### PR TITLE
Reorder columns new implementation

### PR DESCRIPTION
**Pull request by stevekogo**

### CODE REVIEW

**Comment by stevekogo** [comment]
@dannyparsons this is working for now but ucrReorder needs to inherit from ucrCore to enable complete re-implementation

**Comment by Lunalo** [comment]
@stevekogo 

We implemented that yesterday, It was brilliant for you not to edit it

**Comment by maxwellfundi** [comment]
@stevekogo  we have been having that issue with one of the @stevenndungu dialogs.  We fixed that already

**Comment by stevekogo** [comment]
This is not yet complete since we need to update the ucrReorder control to cater to column names appropriately but this is going to wait untill @stevenndungu is done editing it in #2606

**Comment by dannyparsons** [comment]
If everything else in the dialog is functional then it can be accepted and redone when work on other controls has been done. Can a reviewer check this? @africanmathsinitiative/developers

File: instat/dlgReorderColumns.vb [hunk 1]
```diff
@@ -1,61 +1,93 @@
...
+    End Sub
+
+    Private Sub ucrBase_ClickReset(sender As Object, e As EventArgs) Handles ucrBase.ClickReset
+        SetDefaults()
+        SetRCodeForControls(True)
+        TestOkEnabled()
+    End Sub
+
+    Private Sub ucrReorderColumns_OrderChanged() Handles ucrReorderColumns.OrderChanged
         If Not ucrReorderColumns.isEmpty Then
```
Comment (lilyclements) [inline instat/dlgReorderColumns.vb]: Surely you can now do ucrReorderColumns.SetParameter("",) now in initialise?

**Comment by stevekogo** [comment]
@africanmathsinitiative/developers This is ready for review.

File: instat/dlgReorderColumns.vb [hunk 2]
```diff
@@ -1,61 +1,84 @@
...
-        Else
-            ucrBase.OKEnabled(False)
-        End If
-    End Sub
-
-    Private Sub ucrDataFrameSelect_DataFrameChanged(sender As Object, e As EventArgs, strPrevDataFrame As String) Handles ucrDataFrameSelect.DataFrameChanged
-        ucrBase.clsRsyntax.AddParameter("data_name", Chr(34) & ucrDataFrameSelect.cboAvailableDataFrames.Text & Chr(34))
-        TestOkEnabled()
-    End Sub
+            ucrBase.OKEnabled(True)
```
Comment (lilyclements) [inline instat/dlgReorderColumns.vb]: What if the data frame is empty?

**Comment by lilyclements** [comment]
@stevekogo how is this branch going?

**Comment by stevekogo** [comment]
I had a problem with resetting the re-order control. It still kept the ordered values even after the dataframe has been deleted. I am going to re-visit this soon.

**Pull request closed without merging**